lib/Sema/CSSimplify.cpp

```diff
@@ -9249,6 +9249,40 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
     }
   }
 
+  // Special handling of injected references to `makeIterator` in for-in loops.
+  {
+    auto memberRef = getAsExpr<UnresolvedDotExpr>(locator->getAnchor());
+    if (memberRef && memberRef->isImplicit() &&
+        locator->isLastElement<LocatorPathElt::Member>()) {
+      // Cannot simplify this constraint yet since we don't know whether
+      // the base type is going to be existential or not.
+      if (baseObjTy->isTypeVariableOrMember())
+        return formUnsolved();
+
+      auto *sequenceExpr = memberRef->getBase();
+      // If base type is an existential, member lookup is fine because
+      // it would return a witness.
```

Comment:
Member lookup on an existential type doesn't necessarily return a witness - it could also find a `makeIterator` overload in a protocol extension. Let's not try to change that in this PR though.

Reply:
Can we make a note of it in the comment here and track it as an issue?

Reply:
I'd like to make it right so we don't have to worry about it, added changes to fix it.